BBallStatsTrack/src/com/example/bballstatstrack/models/Game.java

```diff
@@ -0,0 +1,496 @@
+package com.example.bballstatstrack.models;
+
+import java.util.Date;
+import java.util.UUID;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.example.bballstatstrack.models.game.GameLog;
+import com.example.bballstatstrack.models.gameevents.BlockEvent;
+import com.example.bballstatstrack.models.gameevents.GameEvent;
+import com.example.bballstatstrack.models.gameevents.GameEvent.ReboundType;
+import com.example.bballstatstrack.models.gameevents.GameEvent.ShotType;
+import com.example.bballstatstrack.models.gameevents.ReboundEvent;
+import com.example.bballstatstrack.models.gameevents.ShootEvent;
+import com.example.bballstatstrack.models.gameevents.TurnoverEvent;
+import com.example.bballstatstrack.models.gameevents.foulevents.NonShootingFoulEvent;
+import com.example.bballstatstrack.models.gameevents.foulevents.ShootingFoulEvent;
+import com.example.bballstatstrack.models.utils.GameEventDeserializer;
+
+import android.text.format.DateFormat;
+import android.util.Log;
+import android.util.SparseArray;
+
+public class Game
+{
+    private static final String B_BALL_STAT_TRACK = "BBallStatTrack";
+
+    private static final int MAX_SHOT_CLOCK = 24;
+
+    private Team mAwayTeam;
+
+    private Team mHomeTeam;
+
+    private Team mHasBallPossession;
```

Comment:
Agrees with Game should handle ball possession but before mHasBallPossession was used to only check which team has the ball and swap and I think that using Team object would make the code look complicated. 

Now with the updated file, it is used to update time. This is ok.